runtime/near-vm-runner/src/tests/ts_contract.rs

```diff
@@ -23,10 +24,11 @@ pub fn test_ts_contract() {
         let promise_results = vec![];
         let runtime = vm_kind.runtime(config.clone()).expect("runtime has not been compiled");
         let result = runtime.run(
-            &code,
+            *code_hash,
+            Some(&code),
```

Comment:
Not sure what you mean by this? Is it basically a first iteration of the `impl FnOnce() -> ContractCode` idea? (which does sound like a great idea :) even though in practice there is barely any work duplicated with the current Option-based version)

Reply:
Yeah this is the first iteration of the closure thought. And also what I was trying to suggest when writing out the idea on Zulip. Intent and the result is the same either way though -- the runtime gets to decide when it needs the wasm code and it can decide at the last moment.